pkg/service/healthserver/healthserver.go

```diff
@@ -189,6 +190,13 @@ func (h *httpHealthHTTPServerFactory) newHTTPHealthServer(port uint16, svc *Serv
 
 		if err := srv.ListenAndServe(); err != http.ErrServerClosed {
 			svc := srv.loadService()
+			if strings.Contains(err.Error(), "bind: address already in use") {
```

Comment:
Is it possible to do a better matching on the error? Have you checked whether `net/http.ListenAndServe()` has a special error type for `addr already in use`?

Reply:
Couldn't find a better option.

Reply:
`net/http.ListenAndServe()` returns an error if `net.Listen()` fails, and the latter should return `syscall.Errno`, so perhaps you could check for `syscall.EADDRINUSE`?